### PR TITLE
[zh-cn]: update the translation of `insertAdjacentElement()`

### PR DESCRIPTION
**Pull request by xiyuesaves**
Fix some translation issues and adjust example code.

### Description

Correct machine translation errors.

Update example code; the reference links in the original example code are no longer valid.

### Additional details

Some content is referenced from [element.insertAdjacentHTML](https://developer.mozilla.org/zh-CN/docs/Web/API/Element/insertAdjacentHTML)

### CODE REVIEW

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 1]
```diff
@@ -5,40 +5,38 @@ slug: Web/API/Element/insertAdjacentElement
```
Comment (JasonLamv-t) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:5]: Insert meta in line#4:
```md
l10n:
  sourceCommit: bbf7f25f9cf95fb154e2740a9fdc9c02818981bf
```

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 2]
```diff
@@ -5,40 +5,38 @@ slug: Web/API/Element/insertAdjacentElement
 
 {{APIRef("DOM")}}
 
-**`insertAdjacentElement()`** 方法将一个给定的元素节点插入到相对于被调用的元素的给定的一个位置。
+**`insertAdjacentElement()`** 方法将一个给定的 {{domxref("Element")}} 元素插入到 DOM 树中的指定位置。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:8]:
```suggestion
{{domxref("Element")}} 接口的 **`insertAdjacentElement()`** 方法将一个给定的元素节点插入到相对于当前元素的指定位置。
```

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 3]
```diff
@@ -5,40 +5,38 @@ slug: Web/API/Element/insertAdjacentElement
...
 
 ## 语法
 
 ```
 element.insertAdjacentElement(position, element);
```
Comment (JasonLamv-t) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:13]:
```suggestion
```js-nolint
insertAdjacentElement(position, element)
```

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 4]
```diff
@@ -5,40 +5,38 @@ slug: Web/API/Element/insertAdjacentElement
...
 ```
 
-### 参数
-
 - position
 
-  - : A {{domxref("DOMString")}} 表示相对于该元素的位置；必须是以下字符串之一：
+  - : 一个 {{domxref("DOMString")}}，表示插入内容相对于元素的位置，并且必须是以下字符串之一：
```
Comment (JasonLamv-t) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:18]:
```suggestion
  - : 表示相对于 `targetElement` 的位置的字符串；必须是以下字符串之一（大小写不敏感）：
```

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 5]
```diff
@@ -5,40 +5,38 @@ slug: Web/API/Element/insertAdjacentElement
...
 
-    - `'beforebegin'`: 在该元素本身的前面。
-    - `'afterbegin'`:只在该元素当中，在该元素第一个子孩子前面。
-    - `'beforeend'`:只在该元素当中，在该元素最后一个子孩子后面。
-    - `'afterend'`: 在该元素本身的后面。
+    - `'beforebegin'`：元素自身的前面。
+    - `'afterbegin'`：插入元素内部的第一个子节点之前。
+    - `'beforeend'`：插入元素内部的最后一个子节点之后。
+    - `'afterend'`：元素自身的后面。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:23]:
```suggestion
    - `'beforebegin'`：`targetElement` 之前。
    - `'afterbegin'`：`targetElement` 内部的第一个子节点之前。
    - `'beforeend'`：`targetElement` 内部的最后一个子节点之后。
    - `'afterend'`：`targetElement` 之后。
```

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 6]
```diff
@@ -5,40 +5,38 @@ slug: Web/API/Element/insertAdjacentElement
...
 
 - element
-  - : 要插入到树中的元素。
+  - : 要被插入到 DOM 树中的 {{domxref("Element")}} 元素。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:26]:
```suggestion
  - : 要插入到树中的元素。
```

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 7]
```diff
@@ -5,40 +5,38 @@ slug: Web/API/Element/insertAdjacentElement
...
+插入的元素，插入失败则返回 `null`。
 
 ### 异常
 
-| 异常          | 说明                           |
-| ------------- | ------------------------------ |
-| `SyntaxError` | 插入的位置是一个无法识别的值。 |
-| `TypeError`   | 插入的元素不是一个有效元素。   |
+- `SyntaxError` {{domxref("DOMException")}}
+  - 无法识别 `position` 的值。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:35]:
```suggestion
  - 如果指定的 `position` 不是可识别的值，则抛出该异常。
```

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 8]
```diff
@@ -5,40 +5,38 @@ slug: Web/API/Element/insertAdjacentElement
...
+- {{jsxref("TypeError")}}
+  - 插入的 `element` 不是一个有效元素。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:37]:
```suggestion
  - 如果指定的 `element` 不是有效的元素，则抛出该异常。
```

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 9]
```diff
@@ -50,28 +48,21 @@ element.insertAdjacentElement(position, element);
 <!-- afterend -->
 ```
 
-> **备注：** 当节点处于 DOM 树中而且有一个父元素的时候 `beforebegin` 和 `afterend` 操作才能起作用。
+> **备注：** 只有当节点处于 DOM 树中且有一个父元素时 `beforebegin` 和 `afterend` 才会生效。
 
 ## 例子
```
Comment (JasonLamv-t) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:53]:
```suggestion
## 示例
```

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 10]
```diff
@@ -50,28 +48,21 @@ element.insertAdjacentElement(position, element);
...
-  setListener(tempDiv);
-});
+// 原为 <div id="one">one</div>
+const one = document.getElementById("one");
+const two = document.createElement("div");
+two.innerText = "two";
+one.insertAdjacentHTML("afterend", two);
+
+// 此时，新结构变为：
+// <div id="one">one</div><div id="two">two</div>
```
Comment (JasonLamv-t) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:63]: If u want to update the example, pls update the upstream(mdn/content) first.
```suggestion
beforeBtn.addEventListener("click", () => {
  const tempDiv = document.createElement("div");
  tempDiv.style.backgroundColor = randomColor();
  if (activeElem) {
    activeElem.insertAdjacentElement("beforebegin", tempDiv);
  }
  setListener(tempDiv);
});

afterBtn.addEventListener("click", () => {
  const tempDiv = document.createElement("div");
  tempDiv.style.backgroundColor = randomColor();
  if (activeElem) {
    activeElem.insertAdjacentElement("afterend", tempDiv);
  }
  setListener(tempDiv);
});
```

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 11]
```diff
@@ -50,28 +48,21 @@ element.insertAdjacentElement(position, element);
...
 ```
 
-看看我们在 Github（也可以参考 [源码](https://github.com/mdn/insert-adjacent/blob/master/insertAdjacentElement.html)）上的 [insertAdjacentElement.html](http://mdn.github.io/insert-adjacent/insertAdjacentElement.html) 演示。在一个容器当中我们有一组{{htmlelement("div")}} 元素。点击其中一个 div 时，这个容器会处于选中状态，然后你就可以按下*Insert before* 或*Insert after* 按钮通过 `insertAdjacentElement() 方法`来把一个新的 divs 插入到选中的元素前面或者后面。
-
```
Comment (JasonLamv-t) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:74]: The demo was removed, I recommend updating the upstream. /cc @yin1999

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 12]
```diff
@@ -85,4 +76,4 @@ afterBtn.addEventListener("click", function () {
 - {{domxref("Element.insertAdjacentHTML()")}}
 - {{domxref("Element.insertAdjacentText()")}}
 - {{domxref("Node.insertBefore()")}}
```
Comment (JasonLamv-t) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:78]:
```suggestion
- {{domxref("Node.insertBefore()")}}（与 `beforebegin` 类似，但参数不同）
```

**Review by JasonLamv-t — CHANGES_REQUESTED** [review]

File: files/zh-cn/web/api/element/insertadjacentelement/index.md (hunk 11 above)
Comment (yin1999) [reply files/zh-cn/web/api/element/insertadjacentelement/index.md:74]: The demo is listed in upstream: https://github.com/mdn/content/blob/4b600e6fa6ccae366aa6c885240f60b748c05c05/files/en-us/web/api/element/insertadjacentelement/index.md?plain=1#L90-L94

What do you mean here?

File: files/zh-cn/web/api/element/insertadjacentelement/index.md (hunk 7 above)
Comment (yin1999) [reply files/zh-cn/web/api/element/insertadjacentelement/index.md:35]: Try to fix the DFL in upstream: mdn/content#32317

File: files/zh-cn/web/api/element/insertadjacentelement/index.md (hunk 11 above)
Comment (JasonLamv-t) [reply files/zh-cn/web/api/element/insertadjacentelement/index.md:74]: Oops, that is my fault. I tried directly to visit the link but got 404. I didn't realize the link was updated in the original text.

Hi, @xiyuesaves. Please add the line#73 back and update the link and the translation. 😊

Comment (xiyuesaves) [reply files/zh-cn/web/api/element/insertadjacentelement/index.md:74]: > Oops, that is my fault. I tried directly to visit the link but got 404. I didn't realize the link was updated in the original text.
> 
> Hi, @xiyuesaves. Please add the line#73 back and update the link and the translation. 😊

Okay, thank you for your help 😊. I'm still not very familiar with this process.

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 13]
```diff
@@ -1,44 +1,44 @@
...
 ### 异常
 
-| 异常          | 说明                           |
-| ------------- | ------------------------------ |
-| `SyntaxError` | 插入的位置是一个无法识别的值。 |
-| `TypeError`   | 插入的元素不是一个有效元素。   |
+- `SyntaxError` {{domxref("DOMException")}}
+  - 如果指定的 `position` 不是可识别的值，则抛出该异常。
+- {{jsxref("TypeError")}}
+  - 如果指定的 `element` 不是有效的元素，则抛出该异常。
```
Comment (yin1999) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:39]: Fixed in upstream, please follow the changes:

```suggestion
  - : 如果指定的 `position` 不是可识别的值，则抛出该异常。
- {{jsxref("TypeError")}}
  - : 如果指定的 `element` 不是有效的元素，则抛出该异常。
```

Comment (xiyuesaves) [reply files/zh-cn/web/api/element/insertadjacentelement/index.md]: Okay, sorry, I just saw the email now.

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 14]
```diff
@@ -1,44 +1,44 @@
 ---
 title: Element.insertAdjacentElement()
```
Comment (yin1999) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:2]:
```suggestion
title: Element：insertAdjacentElement() 方法
```

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 15]
```diff
@@ -1,44 +1,44 @@
...
 slug: Web/API/Element/insertAdjacentElement
+l10n:
+  sourceCommit: bbf7f25f9cf95fb154e2740a9fdc9c02818981bf
```
Comment (yin1999) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:5]:
```suggestion
  sourceCommit: 1e2e601d0d05dca2ab93e0e28f904aa92e4102e1
```

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 16]
```diff
@@ -1,44 +1,44 @@
...
 
+```js-nolint
+insertAdjacentElement(position, element)
 ```
-element.insertAdjacentElement(position, element);
-```
-
-### 参数
 
 - position
```
Comment (yin1999) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:18]:
```suggestion
- `position`
```

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 17]
```diff
@@ -1,44 +1,44 @@
...
-    - `'beforebegin'`: 在该元素本身的前面。
-    - `'afterbegin'`:只在该元素当中，在该元素第一个子孩子前面。
-    - `'beforeend'`:只在该元素当中，在该元素最后一个子孩子后面。
-    - `'afterend'`: 在该元素本身的后面。
+    - `'beforebegin'`：`targetElement` 之前。
+    - `'afterbegin'`：`targetElement` 内部的第一个子节点之前。
+    - `'beforeend'`：`targetElement` 内部的最后一个子节点之后。
+    - `'afterend'`：`targetElement` 之后。
 
 - element
```
Comment (yin1999) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:27]:
```suggestion
- `element`
```

File: files/zh-cn/web/api/element/insertadjacentelement/index.md [hunk 18]
```diff
@@ -50,27 +50,31 @@ element.insertAdjacentElement(position, element);
...
-  activeElem.insertAdjacentElement("afterend", tempDiv);
+  if (activeElem) {
+    activeElem.insertAdjacentElement("afterend", tempDiv);
+  }
   setListener(tempDiv);
 });
 ```
 
-看看我们在 Github（也可以参考 [源码](https://github.com/mdn/insert-adjacent/blob/master/insertAdjacentElement.html)）上的 [insertAdjacentElement.html](http://mdn.github.io/insert-adjacent/insertAdjacentElement.html) 演示。在一个容器当中我们有一组{{htmlelement("div")}} 元素。点击其中一个 div 时，这个容器会处于选中状态，然后你就可以按下*Insert before* 或*Insert after* 按钮通过 `insertAdjacentElement() 方法`来把一个新的 divs 插入到选中的元素前面或者后面。
+看看我们在 Github（也可以参考 [源码](https://github.com/mdn/dom-examples/blob/main/insert-adjacent/insertAdjacentElement.html)）上的 [insertAdjacentElement.html](https://mdn.github.io/dom-examples/insert-adjacent/insertAdjacentElement.html) 演示。在一个容器中我们有一组 {{htmlelement("div")}} 元素。点击其中一个 `div` 时，这个容器会处于选中状态，然后你就可以按下 `Insert before` 或 `Insert after` 按钮，通过 `insertAdjacentElement()` 方法来把一个新的 `div` 插入到选中的元素前面或者后面。
```
Comment (yin1999) [inline files/zh-cn/web/api/element/insertadjacentelement/index.md:77]:
```suggestion
看看我们在 Github（也可以参考[源码](https://github.com/mdn/dom-examples/blob/main/insert-adjacent/insertAdjacentElement.html)）上的 [insertAdjacentElement.html](https://mdn.github.io/dom-examples/insert-adjacent/insertAdjacentElement.html) 演示。在一个容器中我们有一组 {{htmlelement("div")}} 元素。点击其中一个 `div` 时，这个容器会处于选中状态，然后你就可以按下 `Insert before` 或 `Insert after` 按钮，通过 `insertAdjacentElement()` 方法来把一个新的 `div` 插入到选中的元素前面或者后面。
```

**Review by yin1999 — APPROVED** [review]
Thank you @xiyuesaves, @JasonLamv-t

**Pull request merged**